### PR TITLE
Fix ndjson parser to store JSON fields correctly under `target`

### LINKED ISSUE #29395: ndjson parser: json fields are stored not correctly under `target`
Configuration example:
```
parsers:
- ndjson:
    add_error_key: true
    target: kubernetes.audit
```
I would expect to get it stored like:
```
"kubernetes": {
   "audit": {
      "userAgent": ...,
```
but instead I am getting:
```
"kubernetes.audit": {
   "userAgent": ...
    ...
}
```

### PR DESCRIPTION
**Pull request by zipperle**
## What does this PR do?

Stores the JSON fields in a nested way if the `target` of the ndjson parser contains a dotted key, e.g.

the following config:
```
parsers:
- ndjson:
    add_error_key: true
    target: kubernetes.audit
```
leads to this result:
```
"kubernetes": {
   "audit": {
      "userAgent": ...,
```
instead of this:
```
"kubernetes.audit": {
   "userAgent": ...
    ...
}
```

## Why is it important?

Fixes unexpected behavior. 

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## How to test this PR locally

Tested with a unit test.

## Related issues

- Closes #29395

### CODE REVIEW

File: CHANGELOG-developer.next.asciidoc [hunk 1]
```diff
@@ -83,6 +83,7 @@ The list below covers the major changes between 7.0.0-rc2 and main only.
 - Fix the multiple host support for mongodb module {pull}34624[34624]
 - Skip HTTPJSON flakey test. {issue}34929[34929] {pull}35138[35138]
 - Fix ingest pipeline for panw module to parse url scheme correctly {pull}35757[35757]
+- Fix ndjson parser to store JSON fields correctly under `target` {issue}29395[29395]
```
Comment (ycombinator) [inline CHANGELOG-developer.next.asciidoc:86]: Just noticed that this entry should be in `CHANGELOG.next.asciidoc` since it's a user-facing change.

**Comment by ycombinator** [comment]
@zipperle Left a minor comment about the CHANGELOG entry.  Also, I rebased this PR on `main` a couple of times, so you'll want to pull down the PR's branch from your remote first before making any further changes.

**Comment by zipperle** [comment]
@ycombinator thanks for the hint, I moved the changelog entry accordingly.

File: CHANGELOG.next.asciidoc [hunk 2]
```diff
@@ -257,6 +257,8 @@ automatic splitting at root level, if root level element is an array. {pull}3415
 
 *Elastic Logging Plugin*
 
+*Libbeat*
+- Fix ndjson parser to store JSON fields correctly under `target` {issue}29395[29395]
```
Comment (ycombinator) [inline CHANGELOG.next.asciidoc:261]: We don't have a section for libbeat in the CHANGELOG.  Instead we have a section titled "Affecting all Beats".  This entry should go there.  Specifically, this is the line where the entry should go: https://github.com/elastic/beats/blob/4eaa9180deac7c44be3c45f9e1f35d405cbaf439/CHANGELOG.next.asciidoc?plain=1#L77

**Review by ycombinator — APPROVED** [review]
LGTM.  Thanks for the contribution, @zipperle!

**Pull request merged**